### PR TITLE
Add replace script and build pipeline

### PR DESCRIPTION
**Pull request by ivy-cst**

### CODE REVIEW

File: build/replace/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    string(name: 'fileSelector', defaultValue: '', description: 'Bash file selector e.g. folder/**/File.*')
+    string(name: 'sourceBranch', defaultValue: 'master', description: 'The branch to apply this change on e.g. master, release/11.1, release/10.0')
+    string(name: 'newBranch', defaultValue: 'replace', description: 'The name of the new branch, which gets pushed if it is not a dry run')
+    string(name: 'commitMessage', defaultValue: '', description: 'The commit message')
+  }
+
+  stages {
+    stage('replace') {
+      when {
+        expression { !params.sedRegexp.isEmpty() && !params.fileSelector.isEmpty() }
```
Comment (ivy-lli) [inline build/replace/Jenkinsfile:29]: maybe those two should have a default value. now the pipeline will never run even in dryRun mode right?

File: build/replace/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    cron '@midnight'
+  }
+  
+  parameters {
+    booleanParam name: 'dryRun', defaultValue: true, description: 'whether the build should push the changes or not'
+    string(name: 'sedRegexp', defaultValue: '', description: 'The sed replacement: s#search#replace#g')
+    string(name: 'fileSelector', defaultValue: '', description: 'Bash file selector e.g. folder/**/File.*')
+    string(name: 'sourceBranch', defaultValue: 'master', description: 'The branch to apply this change on e.g. master, release/11.1, release/10.0')
+    string(name: 'newBranch', defaultValue: 'replace', description: 'The name of the new branch, which gets pushed if it is not a dry run')
+    string(name: 'commitMessage', defaultValue: '', description: 'The commit message')
```
Comment (ivy-lli) [inline build/replace/Jenkinsfile:23]: not sure how much value it brings to have these two parameters changeable... IMO it is fine if they are autogenerated by the job, like it is on the other pipelines. But ok for me

**Review by ivy-lli — APPROVED** [review]

File: build/replace/Jenkinsfile (hunk 2 above)
Comment (ivy-cst) [reply build/replace/Jenkinsfile:23]: Normally it has a good reason why we search replace something. Only the sed search replace value can look very complicated, not sure how you want to auto generate it :thinking:

**Comment by alexsuter** [comment]
:+1: 

Alternative approach would be to implement this in java. So you wouldn't have to maintain a list of git repos here, too.

**Comment by ivy-cst** [comment]
> 👍
> 
> Alternative approach would be to implement this in java. So you wouldn't have to maintain a list of git repos here, too.

Yes I know. But was easier to reuse the raiseRepo.sh, and was very helpfull for the other jobs to also use the improved raiseRepo.sh ;).

**Pull request merged**